inst/tests/tests.Rraw

```diff
@@ -16846,3 +16846,7 @@ A = data.table(A=c(complex(real = 1:3, imaginary=c(0, -1, 1)), NaN))
 test(2138.3, rbind(A,B), data.table(A=c(as.character(A$A), B$A)))
 A = data.table(A=as.complex(rep(NA, 5)))
 test(2138.4, rbind(A,B), data.table(A=c(as.character(A$A), B$A)))
+
+# column names shouldn't print even if wrapping, #4270
+DT = setDT(replicate(getOption('width'), 1, simplify = FALSE))
+test(2139, {print(DT, col.names='none'); TRUE}, notOutput="v")
```

Comment:
@jangorecki, I had to add the awkward extra `TRUE` because of what seems to be a bug in `test`, I think related to double-`print`ing in this line:

```
out = capture.output(print(x <- suppressMessages(withCallingHandlers(tryCatch(x, error=eHandler), warning=wHandler, message=mHandler))))
```

the `tryCatch(x)` is doing `print` with the correct options, but the `print` in `test()` doesn't keep those options. Then the test fails because the inner `print()` returns `DT`, so the outer print just runs `print` again but without `col.names='none'`, so the test fails. Any idea for a fix? Maybe add `...` option to `test` compatible with `output`/`notOutput` that gets passed to `print`, then we could write the test as:

```
test(2139, DT, notOutput = 'v', col.names = 'none')
```

?

Reply:
yes, I am aware of it... no idea about fixing